Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -1984,20 +1986,24 @@ void MatrixWorkspace::rebuildDetectorIDGroupings() {
     for (const auto &index : (*specDefs)[i]) {
       const size_t detIndex = index.first;
       const size_t timeIndex = index.second;
-      if (detIndex >= allDetIDs.size())
-        throw std::invalid_argument("MatrixWorkspace: SpectrumDefinition "
-                                    "contains an out-of-range detector index, "
-                                    "i.e., the spectrum definition does not "
-                                    "match the instrument in the workspace.");
-      if (timeIndex >= detInfo.scanCount(detIndex))
-        throw std::invalid_argument(
-            "MatrixWorkspace: SpectrumDefinition contains an out-of-range time "
-            "index for a detector, i.e., the spectrum definition does not "
-            "match the instrument in the workspace.");
-      detIDs.insert(allDetIDs[detIndex]);
+      if (detIndex >= allDetIDs.size()) {
+        parallelException = true;
+        error = "MatrixWorkspace: SpectrumDefinition contains an out-of-range "
+                "detector index, i.e., the spectrum definition does not match "
+                "the instrument in the workspace.";
+      } else if (timeIndex >= detInfo.scanCount(detIndex)) {
+        parallelException = true;
+        error = "MatrixWorkspace: SpectrumDefinition contains an out-of-range "
+                "time index for a detector, i.e., the spectrum definition does "
+                "not match the instrument in the workspace.";
+      } else {
+        detIDs.insert(allDetIDs[detIndex]);
+      }
     }
     spec.setDetectorIDs(std::move(detIDs));
   }
+  if (parallelException)
```

Comment:
Looks OK. code `if` and `else if` looks identical. Could probably be refactored out.

Reply:
The error message is actually different.

Reply:
Yep. Didn't spot that first time round.